src/dotnet-scaffolding/Microsoft.DotNet.Scaffolding.Core/Logging/AnsiConsoleSink.cs

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System.Diagnostics;
+using Serilog.Core;
+using Serilog.Events;
+using Spectre.Console;
+
+namespace Microsoft.DotNet.Scaffolding.Core.Logging;
+
+public class AnsiConsoleSink : ILogEventSink
+{
+    public void Emit(LogEvent logEvent)
+    {
+        Debugger.Launch();
+        //should be "enabled", "disabled", "", or null. Empty or null will indicate that a tool is not launched by dotnet-scaffold.
+        bool isRedirectedToDotnetScaffold = !string.IsNullOrEmpty(Environment.GetEnvironmentVariable("DOTNET_SCAFFOLD_TELEMETRY_STATE"));
```

Comment:
Tying this to telemetry doesn't feel great. I get the logic of it. But I sort of wonder if we shouldn't add something more explicit to say that we were launched from dotnet-scaffold. I'm not sure what exactly... Another environment variable?

Reply:
yep another environment variable should be ok here.